### PR TITLE
Fixes #36270 - Update UI to reflect needs_publish on CV

### PR DESCRIPTION
**Pull request by sjha4**
#### What are the changes introduced in this pull request?
Update CV UI to reflect needs_publish flag and add icons to versions and notice to publish wizard.
#### Considerations taken when implementing this change?
We have many pages where we avoid reloading the store from API for CV to avoid a refresh of the page. On such components, added a local store variable to reflect needs_publish based on UI actions.
Caveat: Repository level updates like content changes will need a refresh as there's no way to track that on the UI alone.
#### What are the testing steps for this pull request?
Play around with CV versions and check if the needs_publish icon shows up against the latest version. Also test the publish wizard to see if notice explaining that a publish is not needed shows up when needs_publish is false.

### CODE REVIEW

**Comment by ianballou** [comment]
I'm not seeing `needs_publish` propagate to the UI until I fully refresh the page. I tried updating a filter and then immediately publishing.

**Comment by ianballou** [comment]
One thought -- why is the "update available" icon on the version rather than somewhere at the top of the CV? If it will only ever be on the latest content view version, I wonder if the upgrade icon should be by the CV name instead.

**Comment by sjha4** [comment]
> 

That makes sense..Let me run that by Maria and see if we can update that. 👍🏼

**Comment by ianballou** [comment]
> I'm not seeing `needs_publish` propagate to the UI until I fully refresh the page. I tried updating a filter and then immediately publishing.

I think this may only be happening when deleting filters or filter rules rather than creating them.

**Comment by sjha4** [comment]
> I'm not seeing `needs_publish` propagate to the UI until I fully refresh the page. I tried updating a filter and then immediately publishing.



> 

Ya..We have some pages on the UI where for responsiveness we do not refresh CV details upon some operations. I guess it's time to change that and fetch CV details on those operations. Let me check where things will need updating. 👍🏼

**Comment by sjha4** [comment]
@ianballou : This is ready for re-review with the UI updates on needs_publish changes without refresh.

**Comment by sjha4** [comment]
Updated to clear needs_publish on UI on publish. 👍🏼

File: webpack/scenes/ContentViews/Publish/CVPublishForm.js [hunk 1]
```diff
@@ -36,11 +40,31 @@ const CVPublishForm = ({
...
+                __('No available component content view updates') :
+                __('No available repository or filter updates')}
+              actionClose={
+                <AlertActionCloseButton
+                  onClose={() => setNeedsPublishAlertDismissed(true)}
+                />
+            }
+              style={{ marginBottom: '24px' }}
+            >
+              <TextContent>{__('Newly published version will be the same as previous version.')}</TextContent>
```
Comment (ianballou) [inline webpack/scenes/ContentViews/Publish/CVPublishForm.js:59]:
```suggestion
              <TextContent>{__('Newly published version will be the same as the previous version.')}</TextContent>
```

File: webpack/scenes/ContentViews/components/NeedsPublishIcon.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { Tooltip } from '@patternfly/react-core';
+import { ArrowCircleUpIcon } from '@patternfly/react-icons';
+
+const NeedsPublishIcon = ({ composite }) => (
+  <Tooltip
+    position="auto"
+    enableFlip
+    entryDelay={400}
+    content={composite ? __('Available content view version updates.') :
+      __('Available repositories and/or filters updates.')}
```
Comment (ianballou) [inline webpack/scenes/ContentViews/components/NeedsPublishIcon.js:13]:
```suggestion
      __('Publish update available: repositories and/or filters have changed.')}
```
what do you think about something like this instead?

**Review by ianballou — COMMENTED** [review]
some small comments, works well otherwise!

File: webpack/scenes/ContentViews/components/NeedsPublishIcon.js (hunk 2 above)
Comment (sjha4) [reply webpack/scenes/ContentViews/components/NeedsPublishIcon.js]: Updated. 👍🏼

**Review by ianballou — APPROVED** [review]
This is working for me now!

**Comment by sjha4** [comment]
Merging! 🎉

**Pull request merged**